feature_engine/creation/binary_combination_feature_reference.py

```diff
@@ -0,0 +1,250 @@
+                "corresponding to the names of the variables to be used as  "
+                "reference to combine with the binary operations."
+            )
+        
+        if not isinstance(variables_to_combine, list) or not all(
+            isinstance(var, (int, str)) for var in variables_to_combine
+        ):
+            raise ValueError(
+                "variables_to_combine takes a list of strings or integers "
+                "corresponding to the names of the variables to combine "
+                "with the binary operations."
+            )
+
+        if new_variables_names:
+            if not isinstance(new_variables_names, list) or not all(
+                isinstance(var, str) for var in new_variables_names
+            ):
+                raise ValueError(
+                    "new_variable_names should be None or a list with the "
+                    "names to be assigned to the new variables created by"
+                    "the mathematical combinations."
+                )
+
+        if binary_operations:
+            if not isinstance(binary_operations, list):
+                raise ValueError("binary_operations parameter must be a list or None")
+
+            if any(
+                operation not in ["sub", "div"]
+                for operation in binary_operations
+            ):
+                raise ValueError(
+                    "At least one of the entered binary_operations is not supported. "
+                    "Choose one or more of ['sub', 'div']"
+                )
+
+        # check input logic
+        if len(reference_variables) <= 0:
+            raise KeyError(
+              "reference_variables requires one or more features to make proper "
+              "transformations."
+          )
+
+        if len(variables_to_combine) <= 0:
+            raise KeyError(
+              "variables_to_combine requires one or more features to make proper "
+              "transformations."
+          )
+
+        if new_variables_names:
+            if len(new_variables_names) != (len(reference_variables)*len(variables_to_combine)*len(binary_operations)):
+                raise ValueError(
+                    "Number of items in new_variables_names must be equal to number of "
+                    "items in Reference_variables * intems in variables to combine * binary operations."
+                    "In other words, the transformer needs as many new variable names"
+                    "as reference variables and binary operations to perform over the variables to "
+                    "combine."
+                )
+
+        self.reference_variables = reference_variables
+        self.variables_to_combine = variables_to_combine
+        self.new_variables_names = new_variables_names
+        self.binary_operations = binary_operations
+
+    def fit(self, X: pd.DataFrame, y: Optional[pd.Series] = None):
+        """
+        Performs dataframe checks.
+
+        Args:
```

Comment:
Parameters
------------

Reply:
Also, X and y without indentation. We need this format for the documentation rendering.